python/src/iceberg/catalog/base.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Tuple
+
+import mmh3  # type: ignore
+
+
+class Namespace:
```

Comment:
I'm wondering if we want this in Python. If I had it to do over, I would not put this class in the Java implementation and would use a list of strings for `TableIdentifier` and `Namespace` instead. The problem is that this doesn't do anything useful, but users have to create instances and import it.

I suggest we use a tuple or list of strings instead.

Reply:
That makes total sense to me! I was on the fence as well but I figured it was an easy enough PR to open. But yeah I agree that where this is used should just take an `Iterable[str]` and we could always have a `validate_namespace` and `validate_table_identifier` utility function.